content/using/id/overview.md

```diff
@@ -7,65 +7,189 @@ template = "doc.html"
+
+However, the problem of proving one message was sent before any later ones
+inverts the problem. You can solve this with newspapers by placing the message
+in the text of the newspaper. However, while reading a newspaper requires no
+central party, writing one does. For a long time, this sort of message was
+always handled by a central party. SSL revocations are managed by a few central
+parties. When you buy property, it's not sufficient to have the previous owner
+sign the title — this must be entered into a central land registry. Otherwise,
+the owner may sell their property to multiple people and there would be no way
+to determine who is the new owner. With the land registry, all you need to do is
+ask the registry which sale happened first, and that's the one that counts.
+
+However, Urbit is yours and it's forever. Trusting central registries
+jeopardizes both. The keen reader will notice that the problem of determining
+which key rotation happened first is exactly the double-spend problem that
+Satoshi solved with his proof of work algorithm for Bitcoin. His first block
+famously includes a newspaper headline to prove he didn't mine the block before
+that date. In a beautiful duality, his own algorithm proves that he didn't mine
+it after that date.[^1]
+
+Some argue that blockchain is only good for money. This is myopic and is
+generally based on the experience that its most valuable application so far has
+been money. However, blockchain is a cryptographic primitive to do what was
+previously impossible: prove that one message was signed before another without
+a central party. Blockchain was discovered by someone trying to create digital
+money, and he needed that primitive, but that doesn't mean that's all it's good
+for.
+
+Thus, we store our PKI data on a blockchain for our second kind of name:
+[planets](/docs/glossary/planet). A planet is a 32-bit address which has key information stored on the
+blockchain.[^2] The owner of a planet may broadcast new PKI data by adding it to
+the blockchain. Any later messages by the old keys will be rejected, and
+everyone on the network will listen to the blockchain for key data. Thus, we
+have global consistency, permanence, and self sovereignty. We know of no other
+solution that can provide these properties.
+
+However, while small individually, the aggregate PKI data for all nodes on the
+network may become very large. This is not an issue for comets because nobody
+needs to store comet keys except for those which they're talking with, and even
+those can be garbage-collected and re-requested and verified. For planets, there
+is a canonical set of keys, and somebody must store that. There are about
+2<sup>32</sup>, or 4 billion planets. If the PKI data is about 100 bytes for
+each planet, this is about 400 GB of data. This may be more than most users wish
+to store, but it's small enough that it would be very cheap for someone to host
+this data for many users[^3].
```

Comment:
```suggestion
<a name="3"></a>
However, while small individually, the aggregate PKI data for all nodes on the
network may become very large. This is not an issue for comets because nobody
needs to store comet keys except for those which they're talking with, and even
those can be garbage-collected and re-requested and verified. For planets, there
is a canonical set of keys, and somebody must store that. There are about
2<sup>32</sup>, or 4 billion planets. If the PKI data is about 100 bytes for
each planet, this is about 400 GB of data. This may be more than most users wish
to store, but it's small enough that it would be very cheap for someone to host
this data for many users<sup><a href="#f3">[3]</a></sup>.
```